src/util.ts

```diff
@@ -61,3 +62,36 @@ const outputChannel = vscode.window.createOutputChannel("Azure App Service");
 export function getOutputChannel(): vscode.OutputChannel {
     return outputChannel;
 }
+
+// Telemetry for the extension
+export function sendTelemetry(eventName: string, properties?: { [key: string]: string; }, measures?: { [key: string]: number; }) {
+    if (reporter) {
+        reporter.sendTelemetryEvent(eventName, properties, measures);
+    }
+}
+
+export function errToString(error: any): string {
+    if (error === null || error === undefined) {
+        return '';
+    }
+
+    if (error instanceof Error) {
+        return JSON.stringify({
+            'Error': error.constructor.name,
+            'Message': error.message
+        });
+    }
+
+    if (error instanceof Array) {
```

Comment:
When is error an Array or object? Will calling JSON.stringify or error.toString result in any PII? If not, I'm good with this